src/App.js

```diff
@@ -9,76 +9,57 @@ function App() {
     card: {
         caption: 'Caption',
         text: 'Text...',
-        checked: false,
-        edited: false
-      },
-    cachedCard: {
-      caption: '',
-      text: ''
-    }
+        checked: false
+    },
+    edited: false
   });
 
   function checkboxChange() {
     setCardState({
       ...cardState,
       card: {
         ...cardState.card,
-        checked: !cardState.card.checked}
+        checked: !cardState.card.checked
+      }
     });
   }
 
-  function editCard() {
+  function editMode() {
     setCardState({
       card: {
         ...cardState.card,
-        edited: true,
-        checked: false,
+        checked: false
       },
-      cachedCard: {
-        caption: cardState.card.caption,
-        text: cardState.card.text
-      }
-    });
-  }
-
-  function changeHandler(event, property) {
-    setCardState({
-      ...cardState,
-      cachedCard: {
-        ...cardState.cachedCard, [property]: event.target.value
-      }
+      edited: true
     });
   }
 
-  function saveEditing() {
+  function saveEditing(x, y) {
     setCardState({
-      ...cardState,
       card: {
         ...cardState.card,
-        edited: false,
-      caption: cardState.cachedCard.caption,
-      text: cardState.cachedCard.text
-      }
-    })
+        caption: x,
+        text: y
```

Comment:
You can  pass back with the whole cachedCard data object. If dozens if fields to save changes for - no updates to bring up then. Also you can copy its contents via spread operator - no separate listing for properties then.